### PR TITLE
Ankit| Stage | Version 3 - plan permission

### PR DESCRIPTION
**Pull request by ankitdubeyoffice**
https://app.clickup.com/t/86cwm1bc1

### CODE REVIEW

File: apps/web-giddh/src/app/gst/filing/header/filing-header.component.scss [hunk 1]
```diff
@@ -94,3 +94,8 @@
         }
     }
 }
+// @media screen and (min-width:992px){
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/gst/filing/header/filing-header.component.scss]: Remove commnet

File: apps/web-giddh/src/app/gst/filing/header/filing-header.component.html [hunk 2]
```diff
@@ -82,34 +82,55 @@
...
                     </button>
+                </li>
+                <li>
                     <button type="button" class="btn btn-sky-blue mr-0 gstr-button" (click)="pushToPortalModal?.show()">
                         <i class="icon-download"></i>
                         {{ commonLocaleData?.app_download }}
                     </button>
                 </li>
 
                 <li>
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/gst/filing/header/filing-header.component.html:71]:
```suggestion
```

File: apps/web-giddh/src/app/gst/filing/header/filing-header.component.html [hunk 3]
```diff
@@ -82,34 +82,55 @@
...
+                            [disabled]="(activeCompany$ | async)?.subscription?.planDetails?.restrictedModules.hasOwnProperty('Tax filing')"
                         >
                             {{
                                 gstAuthenticated ? localeData?.filing?.authenticated : localeData?.filing?.authenticate
                             }}
                         </button>
-                    </div>
+                        </li>
+                    </ng-container>
                 </li>
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/gst/filing/header/filing-header.component.html:86]:
```suggestion
```

File: apps/web-giddh/src/app/gst/filing/header/filing-header.component.html [hunk 4]
```diff
@@ -82,34 +82,55 @@
...
             </ul>
+            <div *ngIf="(activeCompany$ | async)?.subscription?.planDetails?.restrictedModules.hasOwnProperty('Tax filing')" class="row pull-right mr-15">
+                <span class="pull-right d-flex align-items-center column-gap5">
+                    <div class="flex-content">{{ commonLocaleData?.app_no_access_limit_module }}</div>
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/gst/filing/header/filing-header.component.html]:
```suggestion
                    <div>{{ commonLocaleData?.app_no_access_limit_module }}</div>
```

**Review by dvCodeWorld — CHANGES_REQUESTED** [review]

**Review by dvCodeWorld — DISMISSED** [review]

**Review by dvCodeWorld — DISMISSED** [review]

**Review by dilpreetsinghofficial — DISMISSED** [review]

File: apps/web-giddh/src/assets/locale/en.json [hunk 5]
```diff
@@ -752,5 +752,7 @@
     "app_subscription": "Subscription",
     "app_custom_email": "Custom Email",
     "app_custom_email_template":"Custom Email Template",
-    "app_send_voucher_type_whatsapp": "Send [VOUCHER_TYPE] via whatsapp"
+    "app_send_voucher_type_whatsapp": "Send [VOUCHER_TYPE] via whatsapp",
+    "app_no_access_limit_module": "You currently don’t have access to this module. Upgrade your plan to unlock it without any hassle",
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/assets/locale/en.json]:
```suggestion
    "app_no_access_limit_module": "You currently don't have access to this module. Upgrade your plan to unlock it without any hassle",
```

File: apps/web-giddh/src/app/onboarding/onboarding.component.ts [hunk 6]
```diff
@@ -55,7 +55,9 @@ export class OnboardingComponent implements OnInit, AfterViewInit, OnDestroy {
     public isGoCardlessSupportedCountry: boolean = false;
     /** Stores the voucher API version of current company */
     public voucherApiVersion: 1 | 2 = 2;
-
+    /** Holds help documentation url for syncing with Tally */
+    public syncWithTallyHelpDocUrl: string = "https://giddh.com/help/sync-with-tally-1591360375828781";
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/onboarding/onboarding.component.ts]: create constant for this in app constant and use

File: apps/web-giddh/src/app/gst/filing/header/filing-header.component.html [hunk 7]
```diff
@@ -61,105 +45,97 @@
...
                         </button>
-                        <button
-                            class="btn btn-primary authenticate-btn pull-right"
+                    </li>
+                    <li>
+                        <button class="btn btn-primary authenticate-btn pull-right"
                             [ngClass]="{ 'btn-success': gstAuthenticated }"
                             (click)="toggleSettingAsidePane($event, 'TAXPRO')"
-                        >
+                            [disabled]="(activeCompany$ | async)?.subscription?.planDetails?.restrictedModules.hasOwnProperty('Tax filing')">
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/gst/filing/header/filing-header.component.html]: put all modules name in a constant and use at all places

File: apps/web-giddh/src/app/gst/filing/header/filing-header.component.html [hunk 8]
```diff
@@ -61,105 +45,97 @@
...
                             }}
                         </button>
-                    </div>
-                </li>
+                    </li>
+                </ng-container>
             </ul>
+            <div *ngIf="(activeCompany$ | async)?.subscription?.planDetails?.restrictedModules.hasOwnProperty('Tax filing')"
+                class="row pull-right mr-15">
+                <span class="pull-right d-flex align-items-center column-gap5">
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/gst/filing/header/filing-header.component.html]: create a common upgrade button module and use at all places

File: apps/web-giddh/src/app/add-company/add-company.component.html [hunk 9]
```diff
@@ -546,8 +546,11 @@
                                     </div>
                                     <div class="row mt-3">
                                         <div class="col-lg-12">
-                                            <a class="mr-0 add-new" href="javascript:void(0);" (click)="addNewUser()">
-                                                <i class="icon-plus"></i> {{ localeData?.add_user }}
+                                            <a class="mr-0 add-new" href="javascript:void(0);"
+                                                (click)="remainingUsers > (thirdStepForm.get('creatorSuperAdmin').value === 'true') + thirdStepForm.get('permissionRoles').value.length ? addNewUser() : null"
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/add-company/add-company.component.html]: kya logic he ye?

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

**Review by dvCodeWorld — APPROVED** [review]

**Review by dvCodeWorld — DISMISSED** [review]

**Review by dvCodeWorld — APPROVED** [review]

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**